src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-                if (pageRankValue == pageRankResults.end()) {
-                    pageRankLocalstore.insert(std::make_pair(startVid, 0.0));
-                } else {
-                    double value = pageRankValue->second;
-                    pageRankLocalstore.insert(std::make_pair(startVid, value));
-                }
-            }
+            write(sockfd, graphID.c_str(), (graphID).size());
+            instance_logger.log("Sent : Graph ID " + graphID, "info");
 
-            string instanceDataFolderLocation =
-                Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
-            string attributeFilePart = instanceDataFolderLocation + "/" + graphID + "_pgrnk_" + partitionID;
-            ofstream partfile;
-            partfile.open(attributeFilePart, std::fstream::trunc);
-            for (map<long, double>::iterator it = pageRankLocalstore.begin(); it != pageRankLocalstore.end(); ++it) {
-                partfile << to_string(it->first) << "\t" << to_string(it->second) << endl;
-            }
-            partfile.close();
-
-            loop = true;
-            pageRankResults.clear();
-            localGraphMap.clear();
-            pageRankLocalstore.clear();
-            graphDBMapCentralStores.clear();
-            graphDBMapLocalStoresPgrnk.clear();
-            instance_logger.log("Finish : Calculate Local page rank.", "info");
-        } else if (line.compare(JasmineGraphInstanceProtocol::WORKER_PAGE_RANK_DISTRIBUTION) == 0) {
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+            write(sockfd, partition.c_str(), (partition).size());
+            instance_logger.log("Sent : Partition ID " + partition, "info");
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+            write(sockfd, JasmineGraphInstanceProtocol::SEND_FILE_NAME.c_str(),
+                  JasmineGraphInstanceProtocol::SEND_FILE_NAME.size());
+            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::SEND_FILE_NAME, "info");
 
             bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string partitionID = (data);
-            partitionID = Utils::trim_copy(partitionID, " \f\n\r\t\v");
-            instance_logger.log("Received Partition ID: " + partitionID, "info");
+            read(sockfd, data, INSTANCE_DATA_LENGTH);
+            string fileName = (data);
```

Comment:
Why do we need the brackets here?

Reply:
This is not present after wrapping read in next PR

Reply:
Ok